src/test/java/org/springframework/integration/aws/s3/config/xml/AmazonS3OutboundChannelAdapterParserTests.java

```diff
@@ -155,4 +172,13 @@ public String generateFileName(Message<?> message) {
 			return null;
 		}
 	}
-}
+
+	public static class FooAdvice extends AbstractRequestHandlerAdvice {
+
+		@Override
+		protected Object doInvoke(ExecutionCallback callback, Object target, Message<?> message) throws Exception {
+			adviceCalled++;
+			return callback.execute();
+		}
+	}
+}
```

Comment:
m-m-m. I think that formatter just doesn't provide the `New line in the end of file` option. 
But this is important. 
So, please, follow with my recommendations.